freedata_server/adif_udp_logger.py

```diff
@@ -11,16 +11,28 @@ def send_adif_qso_data(config, adif_data):
     server_port (int): Port of the server.
     adif_data (str): ADIF-formatted QSO data.
     """
-    adif_log_host = config['MESSAGES'].get('adif_log_host', '127.0.0.1')
-    adif_log_port = int(config['MESSAGES'].get('adif_log_port', '2237'))
+
+    # If False then exit the function
+    adif = config['QSO_LOGGING'].get('enable_adif_udp', 'False')
+
+    if not adif:
+        return  # exit as we don't want to log ADIF UDP
+
+    adif_log_host = config['QSO_LOGGING'].get('adif_udp_host', '127.0.0.1')
+    adif_log_port = int(config['QSO_LOGGING'].get('adif_udp_port', '2237'))
 
     sock = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
     try:
+
+        # If there is no gridsquare fix this so that it follows the ADIF standard.
+        # FIXME this check should probably be in message_system_db_station/model.py
+        # so this should be considered as a quick fix
+        adif_data = re.sub(r"<GRIDSQUARE:4>----", "<GRIDSQUARE:0>", adif_data)
```

Comment:
Then lets change this in database related Module so we have no needed fixes in Future.